spec/lib/annotate/annotate_models_spec.rb

```diff
@@ -2526,44 +2526,256 @@ def annotate_one_file(options = {})
-                                                'foreign_thing_id',
-                                                'foreign_things',
-                                                'id',
-                                                on_delete: :restrict)
-                             ])
-          @schema_info = AnnotateModels.get_schema_info(klass, '== Schema Info', show_foreign_keys: true)
-          annotate_one_file
+        context 'updating a constraint' do
+          let(:updated_foreign_keys) do
+            [
+              mock_foreign_key('fk_rails_cf2568e89e',
+                               'foreign_thing_id',
+                               'foreign_things',
+                               'id',
+                               on_delete: :restrict)
+            ]
+          end
+
+          context 'when option "format_bare" is true' do
+            let(:format_option) { { format_bare: true } }
+
+            it 'should update foreign key constraint' do
+              klass = mock_class(class_name,
+                                 primary_key,
+                                 original_columns,
+                                 [],
+                                 updated_foreign_keys)
+
+              @schema_info = AnnotateModels.get_schema_info(klass, '== Schema Info', options)
+              annotate_one_file(options)
+
+              expect(File.read(@model_file_name)).to eq("#{@schema_info}#{@file_content}")
+            end
+          end
+
+          context 'when option "format_markdown" is true' do
+            let(:format_option) { { format_markdown: true } }
+
+            it 'should update foreign key constraint' do
+              klass = mock_class(class_name,
+                                 primary_key,
+                                 original_columns,
+                                 [],
+                                 updated_foreign_keys)
+
+              @schema_info = AnnotateModels.get_schema_info(klass, '== Schema Info', options)
+              annotate_one_file(options)
+
+              expect(File.read(@model_file_name)).to eq("#{@schema_info}#{@file_content}")
+            end
+          end
+        end
+      end
+
+      context 'adding a new field' do
+        let(:class_name) { :users }
+        let(:primary_key) { :id }
+        let(:original_columns) do
+          [
+            mock_column(primary_key, :integer),
+            mock_column(:name, :string)
+          ]
+        end
+
+        before do
+          klass = mock_class(class_name, primary_key, original_columns)
+          @schema_info = AnnotateModels.get_schema_info(klass, '== Schema Info', options)
+          annotate_one_file(options)
+
+          # confirm we initialized annotaions in file before checking for changes
+          expect(@schema_info).not_to be_empty
+          expect(File.read(@model_file_name)).to eq("#{@schema_info}#{@file_content}")
+        end
+
+        context 'when option "format_bare" is true' do
```

Comment:
I started going down the [shared examples](https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples) route to reduce the noise here. This snowballed a bit since there are additional places where checking for all formats would be useful. I have [a WIP branch](https://github.com/ctran/annotate_models/compare/develop...tmr08c:tmr08c-shared-examples-model-annotation#diff-d35383c256402bb04c05c0f176b1d182) that I'd like to pick up if it looks like the right direction. 

This PR ended up being bigger than I expected, so I thought it would be best to separate additional changes and conversations.